File: YeahLim/05/Solution.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> d, int budget) {
    
    sort(d.begin(), d.end());
    
    int answer = 0;
    
    for (int i = 0; i < d.size(); i++) {
        answer += d[i];
        
        if (answer == budget) {
            return i + 1;
        }
        else if (answer > budget) {
            return i;
        }
    }
    
    return d.size();
}

File: YeahLim/05/Solution.java
class Solution {
    public boolean solution(int x) {
        
        int num = x;
        
        int sum = 0;
        while (x != 0) {
            System.out.println(x%10);
            sum += x % 10;
            x /= 10;
            System.out.println(x);
        }
        
        return num % sum == 0;
    }
}